armi/physics/neutronics/crossSectionGroupManager.py

```diff
@@ -748,9 +771,16 @@ def interactCoupled(self, iteration):
         --------
         :py:meth:`Assembly <armi.physics.neutronics.latticePhysics.latticePhysics.LatticePhysicsInterface.interactCoupled>`
         """
-        # always run for snapshots to account for temp effect of different flow or power statepoint
-        if self.cs[CONF_RUN_TYPE] == "Snapshots" or self.r.p.timeNode == 0:
-            self.interactBOC(cycle=None)
+        if iteration == 0:
```

Comment:
My earlier comment seems to have disappeared. I'll give the TL/DR version this time. 

I think we should leave the functionality here the same as it was and update the cross sections on each tight coupling iteration for timeNode 0. By only running on the zeroth iteration you are assuming that the temperature distribution won't change. Which I don't think we should assume in the general case.

Reply:
It's a user option; a user could also decide to run on every iteration. To me the old implementation doesn't make that much sense; it would update for every coupled iteration for BOC but then not update at all for any other state point.

Reply:
One slight clarification: the old implementation would update for every coupled iteration for timeNode 0 of every cycle and would not update at all for timeNode >= 1.  
https://github.com/terrapower/armi/blob/d5a74d1cc4754f357f6b7d24f3fd6e6d04e1bb52/armi/physics/neutronics/crossSectionGroupManager.py#L735-L745

Reply:
What is the difference between: 
```
 it would update for every coupled iteration for BOC but then not update at all for any other state point.
```

and

```
the old implementation would update for every coupled iteration for timeNode 0 of every cycle and would not update at all for timeNode >= 1.
```
?

Reply:
> What is the difference between: ....

BOC is technically a different statepoint than timeNode 0. We don't do coupling at BOC. We do coupling on the timeNodes. I just wanted to be clear on that distinction. 

> To me the old implementation doesn't make that much sense; it would update for every coupled iteration for BOC but then not update at all for any other state point.

So the new approach would be either:
1. Update on all iterations where tight coupling is used
    - This is expensive but users can pick and choose where to do coupling via the `cyclesSkipTightCouplingInteraction` setting introduced in #1173. 
3. Only update on the first coupling iteration where tight coupling is used 
    - I don't see the general utility in this as it assumes that the temperature distribution won't change after the zeroth iteration. Which I don't think we should assume in the general case.

If that's the case, I would advocate for the first option and removing the second in the interest of clarity/simplicity. What do you think?

Reply:
I think `firstCoupled` is useful because you could have a spatially uniform temperature distribution from the blueprints on the initial `everyNode` interaction (or when loading from a database of an un-coupled case). On the first coupled iteration, you would be capturing a potentially large change in temperature distribution. The change between those first two states could be orders of magnitude larger than the changes in subsequent coupled iterations.

We are not assuming anything is true in the general case, we're just providing an option to speed up calculations when, in the user's judgement, there is a lot of benefit to be had from updating XS after the zeroth coupled iteration and less benefit to be had after that.

Reply:
> On the first coupled iteration, you would be capturing a potentially large change in temperature distribution. The change between those first two states could be orders of magnitude larger than the changes in subsequent coupled iterations.

I totally agree that you would capture a potentially large change and that it would potentially be larger than subsequent calculations. But I would counter with that to really capture the nonlinearity, you would need to continue updating the cross sections for subsequent iterations. Otherwise you're missing the feedback mechanism, not really resolving the nonlinearity adequately. and providing a false convergence. This feedback could be particularly strong for varying types of reactors. 

But I think it's fine to provide the flexibility. My only request would be to rename `firstCoupled` to `firstCoupledIteration` and provide a description of it's intended use case.